pkg/js/transformer/dsltojson.ts

```diff
@@ -44,6 +46,57 @@ interface Relation {
   typeInfo: RelationTypeInfo;
 }
 
+function parseExpression(
+  rewrites: Userset[] | undefined,
+  operator: RelationDefinitionOperator | undefined,
+): Userset | undefined {
+  let relationDef: Userset | undefined;
+
+  if (!rewrites?.length) {
+    return;
+  }
+  if (rewrites?.length === 1) {
+    relationDef = rewrites[0];
+  } else {
+    switch (operator) {
+      case RelationDefinitionOperator.RELATION_DEFINITION_OPERATOR_OR:
+        relationDef = {
+          union: {
+            child: rewrites,
+          },
+        };
+        break;
+      case RelationDefinitionOperator.RELATION_DEFINITION_OPERATOR_AND:
+        relationDef = {
+          intersection: {
+            child: rewrites,
+          },
+        };
+        break;
+      case RelationDefinitionOperator.RELATION_DEFINITION_OPERATOR_BUT_NOT:
+        let rewrite: Userset | undefined;
+        while (rewrites.length >= 2) {
```

Comment:
This feels a bit weird ...how is this different than OR/AND except there are two keywords instead of one.

Reply:
Yah, think you're right. This is a left over artifact from when I was trying to parse

`relation1 and (relation2 but not relation3 but not relation4)`

but after discussion, we're now enforcing `but not`'s to be require brackets:

`relation1 and ((relation2 but not relation3) but not relation4)`

Either way it isn't right.